daemon/cmd/daemon.go

```diff
@@ -324,7 +324,9 @@ func (d *Daemon) restoreCiliumHostIPs(ipv6 bool, fromK8s net.IP) {
 // then it attempts to clear all IPs from the interface.
 func removeOldRouterState(ipv6 bool, restoredIP net.IP) error {
 	l, err := netlink.LinkByName(defaults.HostDevice)
-	if err != nil {
+	if _, linkNotFoundErr := err.(netlink.LinkNotFoundError); linkNotFoundErr && restoredIP == nil {
```

Comment:
Can `errors.As()` be used here to simplify this?

Reply:
Unfortunately, this is not possible with how netlink [exposes](https://github.com/vishvananda/netlink/blob/c94808a88ba452844600de07ac14ed0ee9badca7/link.go#L1346) `LinkNotFoundError`. I mentioned this to the maintainer of netlink, and they fairly said to "submit a PR" essentially. :)

Reply:
cool :)

Reply:
@pippolo84 Tested my assumptions and asserted that `errors.As()` should still work. I was able to confirm it too. Pushed.